### ISSUE TITLE
Undefined output when used programatically

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-linter#1263]
### Describe the problem and steps to reproduce it:

Write a script [exactly as documented](https://github.com/mozilla/addons-linter#import-linter-api-into-another-nodejs-application) to get the linter output.

### What happened?

Instead of an object defining the result, you get `undefined`.

### What did you expect to happen?

It should return an object with information about the linting errors or maybe it should throw an error?

### Anything else we should know?
(Please include screenshots and any relevant files.)

I'm using **addons-linter@0.19.1**

Here is a `script.js` file I wrote as a test:

````js
const addonsLinter = require('addons-linter');

const sourceDir = '/Users/kumar/dev/addons-linter/tests/fixtures/angular-bad-library.zip';

const linter = addonsLinter.createInstance({
  config: {
    _: [sourceDir],
    logLevel: process.env.VERBOSE ? 'debug' : 'fatal',
    stack: Boolean(process.env.VERBOSE),
    pretty: false,
    warningsAsErrors: false,
    metadata: false,
    output: 'none',
    boring: false,
    selfHosted: false,
    shouldScanFile: (fileName) => true,
  },
  runAsBinary: false,
});

linter.run()
  .then((linterResults) => {
    console.log('linterResults', linterResults);
  })
  .catch((err) => console.error("addons-linter failure: ", err));
````

Here is the result:
````
$ node script.js
linterResults undefined
````

If I comment out the `output` key and set `runAsBinary` to `true` then I get the expected result:

````
$ node script.js
Validation Summary:

errors          1
notices         2
warnings        12

ERRORS: ...
````
